app/javascript/components/dropdowns/Reputation.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useEffect, useState } from 'react'
+import { ReputationIcon } from './reputation/ReputationIcon'
+import { ReputationsChannel } from '../../channels/reputationsChannel'
+import { ReputationMenuItem } from './reputation/ReputationMenuItem'
+import { GraphicalIcon } from '../common'
+import { useDropdown, DropdownAttributes } from './useDropdown'
+import { useIsMounted } from 'use-is-mounted'
+import { useRequestQuery } from '../../hooks/request-query'
+import { QueryStatus } from 'react-query'
+import { useErrorHandler, ErrorBoundary } from '../ErrorBoundary'
+import { Loading } from '../common/Loading'
+
+export type Links = {
+  tokens: string
+}
+
+export type ReputationToken = {
+  externalLink: string
+  iconName: string
+  text: string
+  awardedAt: string
+  value: string
+  isSeen: boolean
+}
+
+type APIResponse = {
+  results: ReputationToken[]
+  meta: {
+    links: {
+      tokens: string
+    }
+    reputationSum: number
+  }
+}
+
+const DEFAULT_ERROR = new Error('Unable to reputation tokens')
```

Comment:
I think there is a word missing:

```suggestion
const DEFAULT_ERROR = new Error('Unable to retrieve reputation tokens')
```